gusto_embedded/src/ReactSDKProvider.tsx

```diff
@@ -1,24 +1,51 @@
+import { HTTPClient } from "./lib/http.js";
 import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
 import { GustoEmbeddedProvider } from "./react-query/index.js";
 import { GustoEmbeddedCore } from "./core.js";
 
 //Reexporting useQueryClient to be available in consumer application without direct react-query dependency
-export { useQueryClient } from "@tanstack/react-query"
+export { useQueryClient } from "@tanstack/react-query";
 
-export function ReactSDKProvider({ url, children }: { url: string, children: React.ReactNode }) {
-    const queryClient = new QueryClient();
-    const gustoClient = new GustoEmbeddedCore({
-        serverURL: url,
-    },)
-    queryClient.setQueryDefaults(['@gusto/embedded-api'], { retry: false })
-    queryClient.setMutationDefaults(['@gusto/embedded-api'], { retry: false })
+export function ReactSDKProvider({
+  url,
+  headers,
+  children,
+}: {
+  url: string;
+  headers?: Record<string, string | number>;
```

Comment:
Is there an existing typescript or speakeasy type for headers we could use?

Reply:
oh good call, there is a `Headers` type in typescript, I'll change it to that. I took this from the type signature of headers in the [react sdk config definition](https://github.com/Gusto/embedded-react-sdk/blob/15174b259c59c49f17bb9fb67c6525814aa07449/src/api/client.ts#L9), so you may have to update that when you bumping up to this version of the client library.